src/utils/JiraTicketStatusFetcher.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace staabm\PHPStanTodoBy\utils;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use Psr\Http\Message\StreamInterface;
+
+final class JiraTicketStatusFetcher implements TicketStatusFetcher
+{
+    private const API_VERSION = 2;
+
+    private Client $client;
+    private string $credentials;
+
+    public function __construct(string $host, string $credentialsFilePath)
+    {
+        $credentials = self::getCredentials($credentialsFilePath);
+
+        $this->credentials = base64_encode($credentials);
+        $this->client = new Client([
+            'base_uri' => $host,
+        ]);
+    }
+
+    public function fetchTicketStatus(string $ticketKey): ?string
+    {
+        $apiVersion = self::API_VERSION;
+
+        try {
+            $response = $this->client->get("/rest/api/$apiVersion/issue/$ticketKey", [
+                'query' => [
+                    'expand' => 'status',
+                ],
+                'headers' => [
+                    'Authorization' => "Basic {$this->credentials}",
+                ]
+            ]);
+        } catch (ClientException $exception) {
```

Comment:
Does this mean the issue with the given jira-key does not exist? If so we could report that for the comment (as it might be a typo)

Reply:
According to API's response message, 404 means

> Issue does not exist or you do not have permission to see it

So I'd say we shouldn't assume it's an error to be reported.